### DISCUSSION TITLE
PathPlanner 2024 - Displaying Paths with Field2d

### DISCUSSION THREAD

**Discussion by jonathandao0** [mjansen4857/pathplanner#436] [General]
In previous versions of PathPlanner, you could display the trajectory in a Field2d object using something like this:
```Java
List<Pose2d> trajectoryPoses = 
    trajectory.getStates().stream()
        .map(state -> state.poseMeters)
        .collect(Collectors.toList()));

m_field2d.getObject("trajectory").setPoses(trajectoryPoses);
```

In testing out the PathPlanner 2024 Beta, I tried modifying the code to the following:
```Java
List<Pose2d> poses =
    path.getAllPathPoints().stream()
        .map(point -> new Pose2d(point.position, point.holonomicRotation))
        .collect(Collectors.toList());

m_field2d.getObject("path").setPoses(poses);
```
This causes the code to crash because `point.holonomicRotation` is null for all points, except for the final one which is set by  `GoalEndState` when a path is generated. Is there a recommended way for generating a `List<Pose2d>` from a Path in the updated library other than setting the rotation to 0 for each point?

**Comment by mjansen4857** [comment]
Your options are either:
1. Do what you are currently doing with 0 as the pose rotation (this doesn't get used when displaying poses as a path anyways)
2. Use the custom logging callbacks to get the active path from the path following commands. [Example](https://github.com/mjansen4857/pathplanner/wiki/Java-Example:-Custom-Logging)
3. Generate a trajectory with the path `new PathPlannerTrajectory(path, currentChassisSpeeds)` then get the poses from the list of states in that.

I'd recommend going with option 2 unless you want to display a path on a field 2d before actually running it, in which case option 1 is probably the cleanest.
